include/lf.h

```diff
@@ -62,14 +62,12 @@ typedef struct {
 } LF_PINBOX;
 
 typedef struct {
-  void * volatile pin[LF_PINBOX_PINS];
+  void * volatile pin[LF_PINBOX_PINS] MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE);
```

Comment:
Proper target for alignment is LF_PINS. Did you make sure that whenever pins are allocated they're actually aligned? E.g. what about this:
```
    /* reserve the space for free() address */
    data= alloc + sizeof(void *);
```

That is ```alloc``` may point to aligned address, while ```alloc + sizeof(void*)``` not.